pkg/operator/ceph/object/objectstore.go

```diff
@@ -205,14 +205,6 @@ func deleteSingleSiteRealmAndPools(objContext *Context, spec cephv1.ObjectStoreS
 // This is used for quickly getting the name of the realm, zone group, and zone for an object-store to pass into a Context
 func getMultisiteForObjectStore(ctx context.Context, clusterdContext *clusterd.Context, spec *cephv1.ObjectStoreSpec, namespace, name string) (string, string, string, error) {
 
-	if spec.IsExternal() {
-		// Currently external cluster with zones/zonegroup/realm are not supported, will be
-		// fixed by https://github.com/rook/rook/issues/6342. So if user does not create
-		// zone/zonegroup, RGW internally creates zone/zonegroup named as `default`, hence
-		// Rook can set this value in the object context fields.
-		return "", "default", "default", nil
-	}
```

Comment:
In https://github.com/rook/rook/issues/6342, it was determined that a multisite context isn't needed for external mode CephObjectStores. The context is only needed for managing an object store, which isn't happening in external mode. We should leave this block, but please update the comment so that we don't get confused in the future. My first 2 sentences here are probably a good comment.

Reply:
@thotz can you explain this point a little more?
So why we are sending default values?
 I have tried to reproduce this issue in 1.8 and 1.9 releases with ceph v17 and not been able to reproduce the issue. For external RGW cluster we are not running any radosgw-admin commands but use rest interface via adminops API. The rest interface does not require any zone/zonegroup/realm information since it sends request directly to rgw server endpoint

Reply:
Yes, that's correct. 
For the script in which we create the adminsops user via radosgw-admin command, hence we need the realm information but for Rook internally this is not required

Reply:
So why even `default` is returned by this function,

If the above function would not be called or used for external cluster?
@thotz

Reply:
This function is called for while building objectstore context or it may be called for external cluster, but for external cluster Rook won't run radosgw-admin commands.
Regarding returning "default value", earlier zone/zonegroup/realm won't create for objectstore if multisite is not configured. Later it changed for all the internal object store zone/zonegroup/realm were created
So the condition here was:
if multisite configured 
    return zone/zonegroup/realm values
else 
    return name of objectstore for all three(single site configs)

but this was invalid for external store, hence returned default values that's it. if you want we can return nil as well